File: Source/JavaScript/model/interactions/ExecuteCommandAction.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { InteractionAction } from './InteractionAction';

/**
 * Submits a modeled command.
 */
export interface ExecuteCommandAction extends InteractionAction {
    command: string;
}

export const ExecuteCommandActionPropertyNames: (keyof ExecuteCommandAction)[] = ['command', 'kind'];

File: Source/DotNET/Model/Interactions/ExecuteCommandAction.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Scene.Model.Interactions;

/// <summary>
/// Submits a modeled command.
/// </summary>
/// <param name="Command">The command to submit.</param>
public record ExecuteCommandAction(string Command) : InteractionAction;
